lib/pgBackRest/Expire.pm

```diff
@@ -224,24 +230,33 @@ sub process
         }
     }
 
-    # Default archive retention if not explicily set
-    if (defined($iFullRetention))
+    # if the archive retention is not explicitly set then determine what it should be set to
+    if (!defined($iArchiveRetention))
     {
-        if (!defined($iArchiveRetention))
+        if ($strArchiveRetentionType eq BACKUP_TYPE_FULL && defined($iFullRetention))
         {
             $iArchiveRetention = $iFullRetention;
         }
-
-        if (!defined($strArchiveRetentionType))
+        elsif ($strArchiveRetentionType eq BACKUP_TYPE_DIFF && defined($iDifferentialRetention))
         {
-            $strArchiveRetentionType = BACKUP_TYPE_FULL;
+            $iArchiveRetention = $iDifferentialRetention;
+        }
+        elsif ($strArchiveRetentionType eq BACKUP_TYPE_INCR)
+        {
+            confess &log(ERROR, &OPTION_RETENTION_ARCHIVE_TYPE . " (${strArchiveRetentionType}) not valid without option ".
+                         &OPTION_RETENTION_ARCHIVE, ERROR_OPTION_INVALID);
         }
     }
 
-    # If no archive retention type is set then exit
-    if (!defined($strArchiveRetentionType))
+&log (INFO, "retention-full: ".(defined($iFullRetention) ? $iFullRetention : 'UNDEF').
+            " retention-diff: ".(defined($iDifferentialRetention) ? $iDifferentialRetention : 'UNDEF').
+            " archive-type: ".(defined($strArchiveRetentionType) ? uc($strArchiveRetentionType) : 'UNDEF').
+            " retention-archive: ".(defined($iFullRetention) ? $iFullRetention : 'UNDEF'));
```

Comment:
You should convert this to a logDebugMisc call if it stays in.  It takes care of all the pesky undef testing.